src/swarm-tui/Swarm/TUI/View.hs

```diff
@@ -1443,15 +1444,26 @@ drawRobotLog :: AppState -> Widget Name
 drawRobotLog s =
   vBox
     [ padBottom (Pad 1) (hBorderWithLabel (txt "Log"))
-    , vBox . F.toList . imap drawEntry $ logEntries
+    , vBox . F.toList . imap drawEntry $ logEntriesToShow
     ]
  where
   logEntries = s ^. gameState . to focusedRobot . _Just . robotLog
+  n = Seq.length logEntries
+
+  -- Only show the most recent entry, and any entries which were
+  -- produced by "say" or "log" commands.  Other entries (i.e. errors
+  -- or command status reports) are thus ephemeral, i.e. they are only
+  -- shown when they are the most recent log entry, but hidden once
+  -- something else is logged.
+  logEntriesToShow = logEntries ^.. traversed . ifiltered shouldShow
```

Comment:
If you promote this to a toplevel function, `getLogEntriesToShow`, then this comment can be surfaced to haddock.  It also encapsulates other local definitions, like `n` and `shouldShow`.